FHIR 101 - Practical Guide.ipynb

```diff
@@ -1497,11 +1499,246 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "hidden": true
+   },
+   "outputs": [],
+   "source": [
+    "fetch('http://hapi.fhir.org/baseR4/Observation?_count=5')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "hidden": true
+   },
+   "source": [
+    "###### Summary\n",
+    "The `_summary` parameter returns a subset of information from the search results, such as only certain fields from each result. It can also be used to count the number of results from a query."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "hidden": true
+   },
+   "outputs": [],
+   "source": [
+    "fetch('http://hapi.fhir.org/baseR4/Observation?_summary=count')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "hidden": true
+   },
+   "source": [
+    "##### Fetch all resources types\n",
+    "Fetching all `StructureDefinitions` will give an overview of what types of resources are in a server."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "hidden": true
+   },
+   "outputs": [],
+   "source": [
+    "fetch('http://hapi.fhir.org/baseR4/StructureDefinition?_summary=count')\n",
+    "fetch('http://hapi.fhir.org/baseR4/StructureDefinition')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "hidden": true
+   },
+   "source": [
+    "#####  Fetch a resource's schema\n",
+    "The `$snapshot` parameter will return the full schema for a resource, including any `snapshot` that a resource inherits. \n",
+    "\n",
+    "```\n",
+    "# Example\n",
+    "GET [base]/StructureDefinition/[id]/$snapshot\n",
+    "```\n",
+    "\n",
+    "For example, say there is a resource that inherits from `Patient` and has a `StructureDefinition` that looks like this:\n",
+    "\n",
+    "```json\n",
+    "{\n",
+    "  \"resourceType\": \"StructureDefinition\",\n",
+    "  \"id\": \"591026\",\n",
+    "  \"name\": \"MyPatient\",\n",
+    "  \"type\": \"Patient\",\n",
+    "  \"baseDefinition\": \"http://hl7.org/fhir/StructureDefinition/Patient\",\n",
+    "  \"differential\": {\n",
+    "    \"element\": [ {\n",
+    "      \"id\": \"Patient.extension.race\",\n",
+    "      ...\n",
+    "    } ]\n",
+    "  }\n",
+    "}\n",
+    "```\n",
+    "\n",
+    "Then `$snapshot` will return the full schema for this resource, combining the `differential` from above with the `snapshot` from `Patient`. This is useful because the `$snapshot` parameter will look at a resource's hierarchy and inheritance, and will merge it together, so there is no need for multiple API requests. "
```

Comment:
We may want to also mention that OperationOutcome doesn't support querying only differential?